### DISCUSSION TITLE
Can't InvokeJS in Platform UNO Web Assembly

### DISCUSSION THREAD

**Discussion by DylanSafeAt** [unoplatform/uno#14836] [Q&A]
### Current behavior

I'm trying to warn the user when pressing refresh or exiting the program that progress may not be saved if they do so, I'm using this command to do that in a previous application I developed a year or so ago: 
```
WebAssemblyRuntime.InvokeJS("window.onbeforeunload = function() {\r\nreturn \"\"\r\n}");
```
I'm now developing on the latest release of platform uno, and i keep getting the error "WebAssemblyRuntime does not exist in the current context". Has there been a change in how to invoke js in wasm in recent updates? If so , how would I go about doing it?

**Comment by jeromelaban** [comment]
Thanks for the question.

`WebAssemblyRuntime.InvokeJS` is available, but it has specifics that make it not available when present in the new template "shared" class library. If you still want to use it, you'll need to inject support for it using an interface available in the "shared" class library, then create the implementation in the `.Wasm` head that will use an implementation of that interface.

You should now be using the `JSImport` technique, which does not need to be platform specific, similar to this discussion: https://github.com/unoplatform/uno/discussions/14111#discussioncomment-7461178
